pkg/grpc/remote_grpc_request_authenticator.go

```diff
@@ -0,0 +1,42 @@
+package grpc
+
+import (
+	"context"
+
+	"github.com/buildbarn/bb-storage/pkg/auth"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+type remoteGrpcRequestAuthenticator struct {
```

Comment:
I'd recommend removing Grpc from the name. That it's for gRPC is already implied based on the package name.